src/meta/processors/jobMan/JobDescription.cpp

```diff
@@ -0,0 +1,177 @@
+/* Copyright (c) 2019 vesoft inc. All rights reserved.
+ *
+ * This source code is licensed under Apache 2.0 License,
+ * attached with Common Clause Condition 1.0, found in the LICENSES directory.
+ */
+
+#include <stdexcept>
+#include <string>
+#include <vector>
+#include <folly/String.h>
+#include <boost/stacktrace.hpp>
+#include "meta/processors/jobMan/JobUtils.h"
+#include "meta/processors/jobMan/JobDescription.h"
+
+#include "kvstore/KVIterator.h"
+namespace nebula {
+namespace meta {
+
+using Status = cpp2::JobStatus;
+
+JobDescription::JobDescription(int32_t id,
+                               std::string cmd,
+                               std::vector<std::string> paras,
+                               Status status,
+                               int64_t startTime,
+                               int64_t stopTime)
+                               : id_(id),
+                                 cmd_(std::move(cmd)),
+                                 paras_(std::move(paras)),
+                                 status_(status),
+                                 startTime_(startTime),
+                                 stopTime_(stopTime) {}
+
+folly::Optional<JobDescription>
+JobDescription::makeJobDescription(folly::StringPiece rawkey,
+                                   folly::StringPiece rawval) {
+    try {
+        auto key = parseKey(rawkey);
+        auto tup = parseVal(rawval);
+
+        auto cmd = std::get<0>(tup);
+        auto paras = std::get<1>(tup);
+        for (auto p : paras) {
+            LOG(INFO) << "p = " << p;
+        }
+        auto status = std::get<2>(tup);
+        auto startTime = std::get<3>(tup);
+        auto stopTime = std::get<4>(tup);
+        return JobDescription(key, cmd, paras, status, startTime, stopTime);
+    } catch(std::exception& ex) {
+        LOG(ERROR) << ex.what();
+    }
+    return folly::none;
+}
+
+std::string JobDescription::jobKey() const {
+    return makeJobKey(id_);
+}
+
+std::string JobDescription::makeJobKey(int32_t iJob) {
+    std::string str;
+    str.reserve(32);
+    str.append(reinterpret_cast<const char*>(JobUtil::jobPrefix().data()),
+                                             JobUtil::jobPrefix().size());
+    str.append(reinterpret_cast<const char*>(&iJob), sizeof(int32_t));
+    return str;
+}
+
+int32_t JobDescription::parseKey(const folly::StringPiece& rawKey) {
+    auto offset = JobUtil::jobPrefix().size();
+    if (offset + sizeof(size_t) < rawKey.size()) {
+        std::stringstream oss;
+        oss << __func__ << ", offset=" << offset << ", rawKey.size()=" << rawKey.size();
+        throw std::range_error(oss.str().c_str());
```

Comment:
Did you catch up the exceptions outside?
If there is no necessary, we don't use exception to pass errors to outside.